File: src/main/java/one/oktw/i18n/api/I18n.java
package one.oktw.i18n.api;

import one.oktw.i18n.api.provider.TranslationStringProvider;
import one.oktw.i18n.api.service.TranslationService;
import org.jetbrains.annotations.NotNull;
import org.spongepowered.api.text.serializer.TextSerializer;

import java.util.Locale;

public interface I18n {
    @NotNull
    Registry getRegistry();

    @NotNull
    TextSerializer getLegacySerializer(@NotNull Locale locale);

    @NotNull
    TextSerializer getSpongeSerializer(@NotNull Locale locale);

    @NotNull
    TextSerializer getPlainTextSerializer(@NotNull Locale locale);

    @NotNull
    TranslationService register(@NotNull String scope, @NotNull TranslationStringProvider provider, boolean allowConflict);
}

File: src/main/kotlin/one/oktw/i18n/api/I18n.kt
package one.oktw.i18n.api

import one.oktw.i18n.api.provider.TranslationStringProvider
import one.oktw.i18n.api.service.TranslationService
import org.spongepowered.api.text.serializer.TextSerializer
import java.util.*

interface I18n {
    val registry: Registry

    // get locale aware legacy formatting code serializer
    fun getLegacySerializer(locale: Locale): TextSerializer
    // get locale aware sponge formatting code text serializer
    fun getSpongeSerializer(locale: Locale): TextSerializer
    // get locale aware plain text serializer
    fun getPlainTextSerializer(locale: Locale): TextSerializer

    fun register(scope: String, provider: TranslationStringProvider): TranslationService
}